Adding eventbridge-pipes-ddbstream-sfn-terraform pattern (#2445)
aws-samples/serverless-patterns: eventbridge-pipes-ddbstream-sfn-terraform/README.md

# DynamoDB Stream to Step Functions with EventBridge Pipes

> This pattern shows how to use Amazon EventBridge Pipes to launch a AWS Step Functions state machine with a message coming from an Amazon DynamoDB stream

Author: Makes more sense, thanks. Changes added.

* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
* [Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/aws-get-started) installed

## Deployment Instructions

1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
    ``` 
    git clone https://github.com/aws-samples/serverless-patterns
    ```
2. Change directory to the pattern directory:
    ```
    cd eventbridge-pipes-ddbstream-sfn-terraform
    ```

> 3. From the command line, initialize terraform to download and install the providers defined in the configuration:

Reviewer: ```suggestion
3. From the command line, initialize Terraform to download and install the providers defined in the configuration:
```

Author: Uppercase now

    ```
    terraform init
    ```
4. From the command line, apply the configuration in the main.tf file:
    ```
    terraform apply
    ```
5. During the prompts:
    * Enter yes
6. Note the outputs from the deployment process. These contain the resource names and/or URLs which are used for testing.

## How it works

> Previously, whenever we needed DynamoDB Streams to send messages to Step Functions based on changes made to your DynamoDB table, we had to implement a Lambda function in between DynamoDB and Step Functions because Amazon DynamoDB streams did not support AWS Step Functions as a target directly.

Reviewer: ```suggestion
Previously, whenever you needed to send DynamoDB record changes through DynamoDB streams to Step Functions, you had to implement an AWS Lambda function to invoke a state machine because Amazon DynamoDB streams did not support AWS Step Functions as a direct target.
```

Author: Makes more sense, thanks. Changes added.

> But at re:Invent 2022, AWS launched a new feature for Amazon EventBridge known as EventBridge Pipes.

Reviewer: ```suggestion
```

Author: Done

> Now, we can directly integrate DynamoDB streams with AWS Step Functions without the need for having an additional Lambda function.

Reviewer: ```suggestion
Now, you can directly integrate DynamoDB streams with AWS Step Functions.
```

Author: Makes more sense, thanks. Changes added.

## Testing

> 1. Stream logs from the Step Functions log group

Reviewer: ```suggestion
1. Stream logs from the Step Functions log group. Replace `<LogGroup Name>` with the `SFNLog` output value.
```

```

> sam logs --cw-log-group <LogGroup Name> --tail

Reviewer: Add note to use SFNLog output in the command.

Author: Sorry, I don't understand.  What do you mean?

Reviewer: I mean add an instruction to use the value from `SFNLog` in the `sam logs ` command (i.e. replace LogGroup Name with the value of SFNLog).

Author: The LogGroup name is dynamically generated using a combination of the Stack name and a randomly generated string. I would like to keep the naming convention generic.

```

2. The EventBridge Pipe is configured to filter on the NationalTeam value of "Argentina"
In another terminal, add an item to the DynamoDB stream which does match the filter.

```bash
aws dynamodb put-item \
    --table-name WorldCup-DB \
    --item PlayerName={S="Lionel Messi"},Nationality={S="Argentina"},GoalsScored={S="1"}
```

> The Step Functions workflow should be invoked and you should see the logs for the new execution.

Reviewer: ```suggestion
The Step Functions state machine is invoked and you should see the logs for the new execution.
```

Author: Changed

Now add an item to the DynamoDB stream which doesn't match the filter.
```bash
aws dynamodb put-item \
    --table-name WorldCupTable \
    --item PlayerName={S="Sergy Gnabry"},Nationality={S="Germany"},GoalsScored={S="1"}
```

> No Step Function workflow should be invoked and you should not see any new logs.

Reviewer: ```suggestion
No Step Function state machine is invoked and you will not see any new logs.
```

Author: Changed
